File: backend/src/main/java/org/jdk/project/dto/platform/PlatformMenuDto.java
package org.jdk.project.dto.platform;

import java.util.List;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class PlatformMenuDto {
  private String title;
  private String path;
  private String icon;
  private String module;
  private String permission;
  private Boolean enabled;
  private Boolean pluginPage;
  private Integer sort;
  private List<PlatformMenuDto> children;
}

File: backend/src/main/kotlin/org/jdk/project/dto/platform/PlatformMenuDto.kt
package org.jdk.project.dto.platform

data class PlatformMenuDto(
    var title: String? = null,
    var path: String? = null,
    var icon: String? = null,
    var module: String? = null,
    var permission: String? = null,
    var enabled: Boolean? = null,
    var pluginPage: Boolean? = null,
    var sort: Int? = null,
    var children: List<PlatformMenuDto>? = null,
)
